src/plugins/hetero/async_infer_request.cpp

```diff
@@ -59,6 +59,8 @@ StatusCode HeteroAsyncInferRequest::Wait(int64_t millis_timeout) {
 
 InferenceEngine::Blob::Ptr HeteroAsyncInferRequest::GetBlob(const std::string& name) {
     CheckState();
+    // need lock when different blobs of same infer request from parallel
+    std::lock_guard<std::mutex> lock(_getBlobMutex);
```

Comment:
Why do we need to have this mutex? Does line `setPointerToSo(_heteroInferRequest->getPointerToSo());` cause the issue?

Reply:
yes, the setPointerToSo will have race condition, when getting different blobs of same infer request in hetero, as in hetero, these blobs may come from different hw. I'll check and add the test cases

Reply:
Can we remove global setsopointer? In this case we don't need any mutexes